feature_engine/transformation/arcsin.py

```diff
@@ -0,0 +1,178 @@
+
+    More details in the :ref:`User Guide <arcsin>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    Attributes
+    ----------
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {inverse_transform}
+
+    transform:
+        Apply the arcsin transformation.
+
+    """
+
+    def __init__(
+        self, variables: Union[None, int, str, List[Union[str, int]]] = None
+    ) -> None:
+
+        self.variables = _check_input_parameter_variables(variables)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn parameters.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+        """
+
+        # check input dataframe
+        X = super()._fit_from_varlist(X)
+
+        # check if the variables is in the correct range
+        if ((X[self.variables_] < 0) | (X[self.variables_] > 1)).any().any():
+            raise ValueError(
+                "Some variables contain values outside the possible range"
+                "can't apply arcsin transformation"
+            )
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Apply the arcsin transformation.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_new: pandas dataframe
+            The dataframe with the transformed variables.
+        """
+
+        # check input dataframe and if class was fitted
+        X = super().transform(X)
+
+        # check if the variables is in the correct range
+        if ((X[self.variables_] < 0) | (X[self.variables_] > 1)).any().any():
+            raise ValueError(
+                "Some variables contain values outside the possible"
+                "range for arcsin, can't apply arcsin transformation"
+            )
+
+        # transform
+        X.loc[:, self.variables_] = np.arcsin(np.sqrt(X.loc[:, self.variables_]))
+
+        return X
+
+    def inverse_transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Convert the data back to the original representation.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_tr: pandas dataframe
+            The dataframe with the transformed variables.
+        """
+        # inverse_transform
+        X.loc[:, self.variables_] = np.sin(X.loc[:, self.variables_])
```

Comment:
we are missing the power transformation to revert the square root. Could you please adjust?

Reply:
done (sorry this was really dumb)